addons/account/wizard/account_advance_payment.py

```diff
@@ -0,0 +1,338 @@
+            'order_id': order.id,
+            'display_type': 'line_section',
+            'is_downpayment': True,
+            'sequence': order.order_line and order.order_line[-1].sequence + 1 or 10,
+        }
+
+        del context
+        return so_values
+
+    def _prepare_down_payment_lines_values(self, order):
+        """ Create one down payment line per tax or unique taxes combination and per account.
+            Apply the tax(es) to their respective lines.
+
+            :param order: Order for which the down payment lines are created.
+            :return:      An array of dicts with the down payment lines values.
+        """
+        self.ensure_one()
+
+        if self.advance_payment_method == 'percentage':
+            ratio = self.amount / 100
+        else:
+            ratio = self.fixed_amount / order.amount_total if order.amount_total else 1
+
+        order_lines = order.order_line.filtered(lambda l: not l.display_type and not l.is_downpayment)
+        base_downpayment_lines_values = self._prepare_base_downpayment_line_values(order)
+
+        tax_base_line_dicts = [
+            line._convert_to_tax_base_line_dict(
+                analytic_distribution=line.analytic_distribution,
+                handle_price_include=False
+            )
+            for line in order_lines
+        ]
+        computed_taxes = self.env['account.tax']._compute_taxes(
+            tax_base_line_dicts)
+        down_payment_values = []
+        for line, tax_repartition in computed_taxes['base_lines_to_update']:
+            account = line['product'].product_tmpl_id.get_product_accounts(
+                fiscal_pos=order.fiscal_position_id
+            ).get(self._account_category())
+            taxes = line['taxes'].flatten_taxes_hierarchy()
+            fixed_taxes = taxes.filtered(lambda tax: tax.amount_type == 'fixed')
+            down_payment_values.append([
+                taxes - fixed_taxes,
+                line['analytic_distribution'],
+                tax_repartition['price_subtotal'],
+                account,
+            ])
+            for fixed_tax in fixed_taxes:
+                # Fixed taxes cannot be set as taxes on down payments as they always amounts to 100%
+                # of the tax amount. Therefore fixed taxes are removed and are replace by a new line
+                # with appropriate amount, and non fixed taxes if the fixed tax affected the base of
+                # any other non fixed tax.
+                if fixed_tax.include_base_amount:
+                    pct_tax = taxes[list(taxes).index(fixed_tax) + 1:]\
+                        .filtered(lambda t: t.is_base_affected and t.amount_type != 'fixed')
+                else:
+                    pct_tax = self.env['account.tax']
+                down_payment_values.append([
+                    pct_tax,
+                    line['analytic_distribution'],
+                    line['quantity'] * fixed_tax.amount,
```

Comment:
Feels a bit dangerous recalculating the taxes by hand, there is a method for that in `account.tax`: `compute_all`.

Maybe we can get @smetl 's opinion on this as he's working on a refactoring of the taxes